checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+      return;
+    }
+
+    String receiverString = receiverAsString((FieldAccessNode) lhs);
+    if (TreeUtils.isConstructor(enclosingMethod)) {
+      // Constructors always create must-call obligations, so there is no need for them to
+      // be annotated.
+      return;
+    }
+    ExecutableElement enclosingMethodElt = TreeUtils.elementFromDeclaration(enclosingMethod);
+    MustCallAnnotatedTypeFactory mcAtf =
+        typeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+
+    List<String> cmcfValues =
+        ResourceLeakVisitor.getCreatesMustCallForValues(enclosingMethodElt, mcAtf, typeFactory);
+
+    if (cmcfValues.isEmpty()) {
+      checker.reportError(
+          enclosingMethod,
+          "missing.creates.mustcall.for",
+          receiverString,
+          ((FieldAccessNode) lhs).getFieldName());
+      return;
+    }
+
+    String checked = "";
+    for (String targetStrWithoutAdaptation : cmcfValues) {
+      String targetStr = null;
+      try {
+        targetStr =
+            StringToJavaExpression.atMethodBody(
+                    targetStrWithoutAdaptation, enclosingMethod, checker)
+                .toString();
+      } catch (JavaExpressionParseException e) {
+        targetStr = targetStrWithoutAdaptation;
+      }
+      if (targetStr.equals(receiverString)) {
+        // This @CreatesMustCallFor annotation matches.
+        return;
+      }
+      if ("".equals(checked)) {
+        checked += targetStr;
+      } else {
+        checked += ", " + targetStr;
+      }
+    }
+    checker.reportError(
+        enclosingMethod,
+        "incompatible.creates.mustcall.for",
+        receiverString,
+        ((FieldAccessNode) lhs).getFieldName(),
+        checked);
+  }
+
+  /**
+   * Gets a standardized name for an object whose field is being re-assigned.
+   *
+   * @param fieldAccessNode a field access node
+   * @return the name of the object whose field is being accessed (the receiver), as a string
+   */
+  private String receiverAsString(FieldAccessNode fieldAccessNode) {
+    Node receiver = fieldAccessNode.getReceiver();
+    if (receiver instanceof ThisNode) {
+      return "this";
+    }
+    if (receiver instanceof LocalVariableNode) {
+
+      return ((LocalVariableNode) receiver).getName();
+    }
+    throw new BugInCF(
+        "unexpected receiver of field assignment: " + receiver + " of type " + receiver.getClass());
+  }
+
+  /**
+   * Finds the actual parameter passed in the {@code @MustCallAlias} position for a call.
+   *
+   * @param callNode callNode representing the call
+   * @return if {@code callNode} invokes a method with a {@code @MustCallAlias} annotation on some
+   *     formal parameter (or the receiver), returns the result of calling {@link
+   *     #removeCastsAndGetTmpVarIfPresent(Node)} on the actual parameter passed in that position.
+   *     Otherwise, returns {@code null}.
+   */
+  private @Nullable Node getMustCallAliasParamVar(Node callNode) {
```

Comment:
I'm not sure what the "Var" in "getMustCallAliasParamVar" means.  I think this should be renamed "getMustCallAliasParamVar", and then comments and variables at call sites need to be changed to use proper argument vs. formal parameter terminology.